src/python/pants/engine/native.py

```diff
@@ -437,7 +437,14 @@ def extern_generator_send(self, context_handle, func, arg):
     c = self._ffi.from_handle(context_handle)
     response = self._ffi.new('PyGeneratorResponse*')
     try:
-      res = c.from_value(func[0]).send(c.from_value(arg[0]))
+      cur_val = c.from_value(arg[0])
+      try:
+        res = c.from_value(func[0]).send(cur_val)
+      except StopIteration:
+        # This occurs when a @rule body ends with a `yield Get(...)` statement (without any
```

Comment:
It could be raised for a lot of other reasons as well... it doesn't feel great to catch it in general like this.

Is this a case we should lint for instead?

Reply:
Done! I couldn't think of a way to identify specifically `yield Get`s and was trying to avoid the lint, but it seems necessary for now. Have added an appropriate error message too.